### PR TITLE
[main] adds target namsepace metadata support

### PR DESCRIPTION
**Pull request by jkandasa**
# Changes
* Adds target namespace metadata support via TektonConfig CR
* fixes: [SRVKP-3593](https://issues.redhat.com/browse/SRVKP-3593) - Ability to manage metadata on targetNamespace
* Sample:
  ```
  spec:
    targetNamespaceMetadata:
      labels:
        label1: value1
        label2: value2
      annotations:
        key1: value1
        key2: value2
  ```
# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Run `make test lint` before submitting a PR
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
Under TektonConfig CR, `spec.targetNamespaceMetadata` allows user to update their custom `labels` and `annotations` to the target namespace.
```

### CODE REVIEW

File: pkg/apis/operator/v1alpha1/common.go [hunk 1]
```diff
@@ -72,14 +73,25 @@ type TektonComponentStatus interface {
 type CommonSpec struct {
 	// TargetNamespace is where resources will be installed
 	// +optional
-	TargetNamespace string `json:"targetNamespace,omitempty"`
+	TargetNamespace         string             `json:"targetNamespace,omitempty"`
+	TargetNamespaceMetadata *NamespaceMetadata `json:"targetNamespaceMetadata,omitempty"`
```
[the remark this answers is not in the document]
Comment (jkandasa) [reply pkg/apis/operator/v1alpha1/common.go:77]: @piyush-garg this metadata should goes in to the specific `targetNamespace`, If we have different namespace in different CR/Component, `targetNamespaceMetadata` is applicable to the `targetNamespace`

[the remark this answers is not in the document]
Comment (jkandasa) [reply pkg/apis/operator/v1alpha1/common.go]: @piyush-garg I removed the `namespace` meta support from common.
Now it is only applicable to `TektonConfig` CR.

**Comment by savitaashture** [comment]
/hold

Reviewing PR

**Comment by savitaashture** [comment]
LGTM

**Pull request merged**